.github/workflows/build-msvc.yml

```diff
@@ -193,8 +202,24 @@ jobs:
           make tests
 
       - name: Install the compiler
+        if: matrix.opam != 'true'
         run: make install
 
+      - name: Create opam switch
+        if: matrix.opam == 'true'
+        env:
+          OPAMSWITCH: ${{ matrix.prefix }}
+        run: |
+          make OPAM_PACKAGE_NAME=ocaml-variants INSTALL_MODE=opam install
+          opam init --bare --yes --disable-sandboxing --auto-setup --cygwin-local-install
+          opam switch create '${{ env.OPAMSWITCH }}' --empty
+          opam pin add --no-action --kind=path ocaml-variants .
+          opam pin add --no-action flexdll flexdll
+          opam pin add --no-action winpthreads winpthreads
+          opam install --yes flexdll winpthreads
+          opam install --yes --assume-built ocaml-variants
+          opam exec -- ocamlc -v
```

Comment:
I suppose that the opam cli is quite stable at this point; but we could pass `--cli=2.4` or set `OPAMCLI` for future-proofing.